### ISSUE TITLE
agent() not exposed

### ISSUE THREAD

**Issue by MystK** [lightsofapollo/superagent-promise#12]
Looks like I can't call superagent.agent() to persist sessions and cookies. Without it, I can't login to a website with redirects. Is there something I'm missing, or is there an alternative to calling superagent.agent()?

**Comment by clayreimann** [comment]
Nope, you're not missing anything. No one has needed it so far (or complained about it's absence at any rate).

Simple fix would be to set `superagent`'s `agent` property after [this line](https://github.com/lightsofapollo/superagent-promise/blob/master/index.js#L57)

**Comment by MystK** [comment]
Hey there, not too sure what to put there. I tried:

    request.agent = require('./agent');

Where I copied agent and dependencies from superagent, but it's not working.

**Comment by clayreimann** [comment]
Can you give an example of how you would use `SuperAgent.agent`?

**Comment by lefnire** [comment]
```javascript
var superagent = require('superagent');
var session = superagent.agent();
session.post('/login').send({username:'username',password:'password'}).end(()=>{
  session.get('/posts').end(...)
})
```

`SuperAgent.agent` spins off a single-session to be used during the navigating lifecycle, especially useful during API mocha-testing. Without using `superagent.agent()`, each request has cleared headers & such, and the second `superagent.get('/posts')` would return `401` (if that route is protected). At least, this is how I understand superagent to work ([see this stackoverflow](http://stackoverflow.com/questions/14001183/how-to-authenticate-supertest-requests-with-passport))?

Ie, it's used for session-based requests. I too vote for this feature

**Comment by clayreimann** [comment]
Ok, I'll have to see how they `Superagent` does it. Sounds like it could be tricky, but not impossible

**Comment by heycalmdown** [comment]
+1

**Comment by yura415** [comment]
+1 Also would be glad to see this feature. It's odd how SuperAgent removed docs of this and example script.

**Comment by chpio** [comment]
:+1: yes please :)

**Comment by smirea** [comment]
any update on this? Still not exposed, still an issue :)

**Comment by julianfresco** [comment]
Damn, ran into this dead end. Looks like I'll need to drop this dead project and upgrade superagent to a later version. They've been [supporting Promises since 2.x](https://github.com/visionmedia/superagent/blob/master/History.md#200-2016-05-29)
